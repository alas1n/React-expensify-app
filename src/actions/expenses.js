import uuid from 'uuid'


// add expenses
export const addExpenses = (
    {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
    } = {}
) => ({
    type: 'ADD_EXPENSES',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
})

// REMOVE EXPENSES
export const removeExpenses = (({ id } = {}) => {
    return {
        type: 'REMOVE_EXPENSE',
        id
    }
})

// EDIT EXPENSES
export const editExpenses = ((id, updates) => {
    return {
        type: 'EDIT_EXPENSE',
        id,
        updates
    }
})