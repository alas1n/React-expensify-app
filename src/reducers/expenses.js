// DEFULT VALUE FOR EXPENSE ARRY
const expneseReducerDefaultState = [];


// EXPENSE REDUCER
export default (state = expneseReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSES':
            return [
                ...state,
                action.expense
            ]
        case 'REMOVE_EXPENSE':
            return state.filter(expense => expense.id !== action.id)
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    }
                } else {
                    return expense
                }
            })
        default:
            return state
    }
}


