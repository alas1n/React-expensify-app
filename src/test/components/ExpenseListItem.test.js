import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../Components/ExpenseListItem'
import expenses from '../fixtures/expenses'


test('should render ExpnesListItem whit Expenses', () => {
	const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)
	expect(wrapper).toMatchSnapshot()
})
