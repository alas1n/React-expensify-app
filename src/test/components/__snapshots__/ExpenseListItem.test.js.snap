// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpnesListItem whit Expenses 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    195
    -
    Thursday, January 1st 1970, 3:30:00 am
  </p>
</div>
`;
